examples/sealed-auctions/src/lib.rs

```diff
@@ -0,0 +1,382 @@
+#![feature(proc_macro_hygiene)]
+
+pub mod types;
+
+extern crate chrono;
+
+use chrono::Utc;
+use std::collections::{HashMap, HashSet}; // `use` statements can go anywhere
+
+#[mantle::service]
+mod service {
+    use crate::types::*;
+
+    #[derive(Service)]
+    pub struct AuctionMarket {
+        /// The item counter
+        itemCounter: usize,
+
+        /// The administrators of this auction mechanism
+        /// Administrators can blacklist users without due cause
+        admins: HashSet<UserId>,
+
+        /// The blacklist of users
+        blacklist: HashSet<UserId>,
+
+        /// The current valuation of the market as the cummulative value
+        /// of all artifacts that have been up for auction
+        marketSize: U256,
+
+        /// The repository of unique artifacts
+        artifacts: HashMap<ItemId, Artifact>,
+
+        /// Active auctions
+        auctions: HashMap<ItemId, Auction>,
+    }
+
+    impl AuctionMarket {
+        /// Constructs a new `AuctionMarket`.
+        pub fn new(ctx: &Context) -> Result<Self> {
+            let mut admins = HashSet::new();
+            admins.insert(ctx.sender());
+
+            let artifacts = HashMap::new();
+            let blacklist = HashSet::new();
+            let auctions = HashMap::new();
+            let itemCounter: usize = 0;
+            let marketSize: U256 = U256::from(0);
+
+            Ok(Self {
+                itemCounter,
+                admins,
+                blacklist,
+                marketSize,
+                artifacts,
+                auctions,
+            })
+        }
+
+        /// Adds a new auction item to the marketplace
+        /// Anyone can sell an item. Once they add an item to the market
+        /// they control the auction through its conclusion
+        pub fn add_item(
+            &mut self,
+            ctx: &Context,
+            value: usize,
```

Comment:
huh, usize is architecture independent. I'm not sure this will get through idl-gen. If it doesn't, I'll just map usize to u32.

Reply:
Is there a way I can check if this gets through idl-gen? I can make that change if it doesn't?

Reply:
Will change `usize` to `u64` or `u32` making it explicitly sized. Noticed the comments later in the review.

Reply:
`usize` -> `u64`